### PR TITLE
Linux64pr

### PR DESCRIPTION
**Pull request by kntroh**
I was verified building and running of all snippets on linux 64-bit (Linux Mint 17.1 64-bit, excluding i386 library, and Mint 17.1 32-bit).

Thanks for JesseKPhillips's work.

### CODE REVIEW

File: src/org/eclipse/swt/internal/c/X.d [hunk 1]
```diff
@@ -6,269 +6,295 @@ module org.eclipse.swt.internal.c.X;
...
-const int GXorInverted = 0xd;
+alias int cint;
+alias uint cuint;
+static if (intptr_t.sizeof == 8) {
+    alias long clong;
+    alias ulong culong;
+} else {
+    alias int clong;
+    alias uint culong;
+}
```
Comment (jacob-carlborg) [inline src/org/eclipse/swt/internal/c/X.d]: Why are not `c_long` defined in `core.stdc.config` used?

File: src/org/eclipse/swt/internal/c/XInput.d [hunk 2]
```diff
@@ -49,12 +60,6 @@ alias XDeviceButtonEvent XDeviceButtonReleasedEvent;
 alias XDeviceButtonEvent XDeviceButtonPressedEvent;
 alias XDeviceKeyEvent XDeviceKeyReleasedEvent;
 alias XDeviceKeyEvent XDeviceKeyPressedEvent;
-alias void function(void *, char *, int, int, char * *) _BCD_func__1398;
-alias int function(void *) _BCD_func__1440;
-alias int function(void *, XErrorEvent *) _BCD_func__1441;
-alias void function(void *, char *, char *) _BCD_func__1515;
-alias int function(void *, char *, char *) _BCD_func__1516;
-alias void function(void *, char *, char *) _BCD_func__1517;
```
Comment (jacob-carlborg) [inline src/org/eclipse/swt/internal/c/XInput.d]: Why are these removed?

File: src/org/eclipse/swt/widgets/Display.d [hunk 3]
```diff
@@ -4182,61 +4186,69 @@ static dchar wcsToMbcs (char ch) {
...
     return widget.windowProc (handle, arg0, arg1, arg2, user_data);
 }
 
+private static extern(C) int windowProcChangeValueFunc(GtkWidget* handle, int scroll, double value1, void* user_data) {
+    version(LOG) getDwtLogger().error( __FILE__, __LINE__,  "Display {}:", __LINE__ ).flush;
+    CallbackData* cbdata = cast(CallbackData*)user_data;
+    Widget widget = cbdata.display.getWidget (handle);
+    if (widget is null) return 0;
+    return widget.gtk_change_value(handle, scroll, value1, user_data);
+}
```
Comment (jacob-carlborg) [inline src/org/eclipse/swt/widgets/Display.d]: What is this, a completely new function?

File: src/org/eclipse/swt/accessibility/AccessibleFactory.d [hunk 4]
```diff
@@ -95,7 +95,7 @@ class AccessibleFactory {
         }
     }
 
-    private this (int /*long*/ widgetType) {
+    private this (int widgetType) {
```
Comment (jacob-carlborg) [inline src/org/eclipse/swt/accessibility/AccessibleFactory.d]: I don't really understand why in some cases `int /*long*/` is replaced with `int` and in some cases the expected `ptrdiff_t`.

File: src/org/eclipse/swt/internal/c/X.d (hunk 1 above)
Comment (kntroh) [inline src/org/eclipse/swt/internal/c/X.d]: Oops, I didn't know it. I'll replaces it.

File: src/org/eclipse/swt/internal/c/XInput.d (hunk 2 above)
Comment (kntroh) [inline src/org/eclipse/swt/internal/c/XInput.d]: I was trying to fix these types, but didn't know the identity of functions. Moreover, They are not used.

File: src/org/eclipse/swt/widgets/Display.d (hunk 3 above)
Comment (kntroh) [inline src/org/eclipse/swt/widgets/Display.d]: Declaration of CHANGE_VALUE event doesn't match windowProcFunc5. If use windowProcFunc5, Occurs Segfault.

See also: https://developer.gnome.org/gtk3/stable/GtkRange.html#GtkRange-change-value

File: src/org/eclipse/swt/accessibility/AccessibleFactory.d (hunk 4 above)
Comment (kntroh) [inline src/org/eclipse/swt/accessibility/AccessibleFactory.d]: They are actually int (gint or GLint or int of Xlib).

I checked a declaration of all function in libraries. And if it is to should be AAA, I replaced it to int.

**Comment by kntroh** [comment]
I updated pull request.

**Comment by jacob-carlborg** [comment]
@kntroh @JesseKPhillips awesome, thanks for all the work.

**Comment by kntroh** [comment]
Thanks for your careful review.

**Comment by JesseKPhillips** [comment]
@kntroh Thanks for finishing this up.
